File: 0392_Is_Subsequence/392.py
class Solution:
    def isSubsequence(self, s: str, t: str) -> bool:
        temp = t
        for ch in s:
            try:
                idx = temp.index(ch)
                temp = temp[idx + 1:]
            except:
                return False
        return True

a = Solution()
print(a.isSubsequence(s = "abc", t = "ahbgdc"))
print(a.isSubsequence(s = "axc", t = "ahbgdc"))

File: 0392_Is_Subsequence/392.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    bool isSubsequence(string s, string t)
    {
        string temp = t;
        for (int i = 0; i < s.size(); i++)
        {
            int idx = temp.find(s[i]);
            if (idx == -1)
                return false;
            temp = temp.substr(idx + 1, temp.size() - idx - 1);
        }
        return true;
    }
};

int main()
{
    Solution a;
    cout << a.isSubsequence("abc", "ahbgdc") << " " << a.isSubsequence("axc", "ahbgdc");
    // string s1 = "hello world", s2 = "lox";
    // int pos = s1.find(s2);
    // cout << pos << "\n";
    return 0;
}